### PR TITLE
fix dynamic island overlap on recieve modal

### PR DESCRIPTION
**Pull request by BrodyHughes**
Fixes APP-950

## What changed (plus any additional context for devs)
- moved to SimpleSheet and changed the route to use similar route settings that we use for other sheets

## Screen recordings / screenshots
Before:
Screenshot 2024-04-26 at 12 39 18 PM
After:
<video src="https://github.com/rainbow-me/rainbow/assets/41711440/ad55b593-39ec-4fcd-9a31-f69d50e6b4df"> </video>

### CODE REVIEW

File: src/screens/ReceiveModal.js [hunk 1]
```diff
@@ -74,12 +63,19 @@ export default function ReceiveModal() {
...
 
   return (
-    <Container backgroundColor="rgba(0,0,0,0.85)" testID="receive-modal">
-      <TouchableBackdrop onPress={goBack} />
-      <Handle />
-      <ColumnWithMargins align="center" margin={24}>
+    <SimpleSheet
+      testID="receive-modal"
+      backgroundColor={'rgba(0,0,0,0.85)'}
+      useAdditionalTopPadding={200}
```
Comment (walmat) [inline src/screens/ReceiveModal.js:73]: i'd make sure this doesn't make android look really weird

File: src/screens/ReceiveModal.js [hunk 2]
```diff
@@ -74,12 +63,19 @@ export default function ReceiveModal() {
...
+      customHeight={IS_ANDROID ? deviceHeight - top : deviceHeight - sharedCoolModalTopOffset}
+      scrollEnabled={false}
+      top={{ custom: safeAreaInsetValues.top + 6 }}
```
Comment (walmat) [inline src/screens/ReceiveModal.js:76]: some of these props could be cleaned up. I don't think top is an accepted prop into the SimpleSheet component

File: src/screens/ReceiveModal.js (hunk 1 above)
Comment (walmat) [reply src/screens/ReceiveModal.js:73]: If we need this prop for Android, probably should do

```
additionalTopPadding={IS_ANDROID ? StatusBar.currentHeight : false}
```

Comment (BrodyHughes) [reply src/screens/ReceiveModal.js:73]: i didn't know we had an android app

Comment (mikedemarais) [reply src/screens/ReceiveModal.js:73]: lmao

**Review by walmat — APPROVED** [review]
smol ting but otherwise 🔥

File: ios/Rainbow.xcodeproj/project.pbxproj
[the remark this answers is not in the document]
Comment (BrodyHughes) [reply ios/Rainbow.xcodeproj/project.pbxproj:1]: oops yes

**Pull request merged**